OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -482,5 +482,17 @@ public static void AddPlayerFlagAndName(ScrollItemWidget template, Player player
 			};
 			playerName.GetColor = () => player.Color.RGB;
 		}
+
+		public static string GetExternalIP(int clientIndex, OrderManager orderManager)
+		{
+			var address = orderManager.LobbyInfo.ClientWithIndex(clientIndex).IpAddress;
+			if (clientIndex == orderManager.LocalClient.Index && address == IPAddress.Loopback.ToString())
+			{
+				if (UPnP.NatDevice != null && UPnP.NatDevice.GetExternalIP().ToString() != null)
```

Comment:
I expect I expect `UPnP.NatDevice.GetExternalIP()` is returning null and not the `ToString()` afterwards.